components/MediaPlayer.vue

```diff
@@ -1,174 +1,401 @@
+};
+const onPointerUpProgressBar = (event: PointerEvent) => {
+  const rect = (event.currentTarget as Element)?.getBoundingClientRect();
+  currentPosition.value =
+    ((event.clientX - rect.left) / rect.width) * currentTrackDuration.value;
+};
 </script>
 
 <template>
-  <div
-    class="fixed bottom-4 right-4 flex w-[400px] flex-col rounded-2xl border border-black-1/10 bg-white-1 p-3 shadow-xl dark:border-white-1/10 dark:bg-black-1"
+  <transition
+    enter-active-class="transition-all duration-200 ease-out"
+    enter-from-class="opacity-0 translate-y-2"
+    leave-active-class="transition-all duration-200 ease-out"
+    leave-to-class="opacity-0 translate-y-2"
   >
-    <div class="flex min-w-0 flex-1 gap-3">
-      <div class="group select-none">
-        <button @click.stop="rewind()">
-          <span class="flex aspect-square w-12 justify-center align-middle">
-            <NuxtIcon
-              name="icon.rewind.large"
-              class="text-3xl group-hover:text-4xl"
-            />
-          </span>
-        </button>
-      </div>
-      <div v-if="hasPrevious" class="group select-none">
-        <button @click.stop="previous()">
-          <span class="flex aspect-square w-12 justify-center align-middle">
-            <NuxtIcon
-              name="icon.previous.track.large"
-              class="text-3xl group-hover:text-4xl"
+    <div
+      v-if="!open"
+      class="group absolute bottom-5 right-5 w-[400px]"
+      @click.stop="open = !open"
+    >
+      <svg
+        class="absolute left-0 top-0 opacity-0 group-hover:opacity-100 group-hover:-left-1 group-hover:-top-1 transition-all duration-200 ease-out"
+        xmlns="http://www.w3.org/2000/svg"
+        width="11"
+        height="11"
+        viewBox="0 0 11 11"
+        fill="none"
+      >
+        <path
+          d="M0 1.5C0 0.671573 0.671573 0 1.5 0H7.37868C8.71504 0 9.38429 1.61571 8.43934 2.56066L2.56066 8.43934C1.61572 9.38428 0 8.71504 0 7.37868V1.5Z"
+          fill="currentColor"
+        />
+      </svg>
+      <div
+        class="shadow-player relative overflow-hidden flex flex-col rounded-2xl bg-background-1 p-3"
+      >
+        <div class="flex gap-4">
+          <div class="aspect-square h-[48px] rounded-md">
+            <ProtectedImage
+              v-if="currentTrack?.meta?.attachedPicture"
+              :src="currentTrack?.meta?.attachedPicture"
             />
-          </span>
-        </button>
-      </div>
-      <div class="flex min-w-0 flex-1 gap-3">
-        <div
-          class="aspect-square h-[48px] shrink-0 overflow-hidden rounded-md bg-background-2"
+          </div>
+          <div class="flex gap-1 w-full flex-col overflow-hidden">
+            <h3
+              class="truncate text-lg font-semibold leading-tight"
+              :title="currentTrack?.title || ''"
+            >
+              {{ currentTrack?.title }}
+            </h3>
+            <div class="truncate text-base leading-snug text-label-2">
```

Comment:
The text color doesn't work as well in dark mode here

![CleanShot 2023-12-03 at 14 36 33@2x](https://github.com/bcc-code/bmm-web/assets/18753964/d9f888b7-830c-42a6-9f61-9af0624c285b)